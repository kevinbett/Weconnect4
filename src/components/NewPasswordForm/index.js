import React from 'react';
import './style.css';


class NewPasswordForm extends React.Component {
  state = {
    message: undefined
  }

  handleNewPassword = async (event) => {
    event.preventDefault();
    const token = window.localStorage.getItem('authToken');
    let auth_token = 'Bearer ' + token;
    const new_password = event.target.new_password.value;
    const apiData = await fetch(`https://weconnect4-heroku.herokuapp.com//reset-password/${token}`, {
          method: "POST",
          headers: {'Accept': 'application/json','Content-Type': 'application/json', 'Authorization': auth_token},
          body: JSON.stringify({
            "new_password": new_password

          })
    });

      const res = await apiData.json();
      console.log(res)
      
      if (res.message === "Password has been successfully changed") {
        window.localStorage.setItem("rMessage", res.message)
        window.location.assign('/login')
      };
      this.setState({message: res.message.message, error: res.error});
    }

  render() {
    return (
    <div id="main-registration-container">
     <div id="register">
        <h3>Enter New Password</h3>
        <form onSubmit= {this.handleNewPassword} >
        <p className="errorMsg">{this.state.message}</p>
        <label></label>
        <input type="password" name="new_password"  />
        <input type="submit" className="button"  value="Submit"/>
        </form>
    </div>
</div>

      );
  }


}


export default NewPasswordForm;