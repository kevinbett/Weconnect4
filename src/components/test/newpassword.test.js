import NewPasswordForm from "../NewPasswordForm"
import { shallow } from "enzyme"
import { Divider } from "../../../node_modules/semantic-ui-react";

describe('New password form',() => { 
    it ('renders new password form', () => { 
        expect(shallow(<NewPasswordForm/>)).toMatchSnapshot()
    });
} )
