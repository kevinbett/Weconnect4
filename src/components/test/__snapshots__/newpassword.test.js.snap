// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New password form renders new password form 1`] = `
<div
  id="main-registration-container"
>
  <div
    id="register"
  >
    <h3>
      Enter New Password
    </h3>
    <form
      onSubmit={[Function]}
    >
      <p
        className="errorMsg"
      />
      <label />
      <input
        name="new_password"
        type="password"
      />
      <input
        className="button"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
